examples/medplum-demo-bots/src/deduplication/merge-matching-patients.ts

```diff
@@ -135,6 +136,40 @@ export function linkPatientRecords(src: Patient, target: Patient): MergedPatient
 
 // end-block linkPatientRecords
 
+// start-block unLinkPatientRecords
+
+/**
+ * Unlink two patient that have been merged
+ *
+ * @param src - The source patient record which is marked as replaced.
+ * @param target - The target patient marked as the master record.
+ * @returns - Object containing updated source and target patient records with their links.
+ */
+export function unLinkPatientRecords(src: Patient, target: Patient): MergedPatients {
+  const targetCopy = deepClone(target);
+  const srcCopy = deepClone(src);
+  // Find the index of the source patient in the target's `link` array
+  const srcIndex = targetCopy.link?.findIndex((link) => resolveId(link.other) === src.id);
+  if (srcIndex !== undefined && srcIndex !== -1) {
+    targetCopy.link?.splice(srcIndex, 1);
```

Comment:
Minor, but the `?` is not needed here because you know `targetCopy.link` is defined.

Reply:
You could also do this as a one liner:

```ts
targetCopy.link = targetCopy.link?.filter((link) => resolveId(link.other) !== src.id);
srcCopy.link = srcCopy.link?.filter((link) => resolveId(link.other) !== target.id);
```

Reply:
Thank you, of course that's nicer